tsconfig.json

```diff
@@ -1,23 +1,30 @@
 {
   "compilerOptions": {
-    // "allowJs": true /* Allow javascript files to be compiled. */,
+    "allowJs": true /* Allow javascript files to be compiled. */,
     // "checkJs": true /* Report errors in .js files. */,
-    "jsx": "react" /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */,
-    "target": "es2017",
-    "module": "commonjs",
-    "declaration": true,
-    "outDir": "build",
+    "outDir": "dist",
+    "module": "esnext",
+    "lib": ["dom", "esnext"],
+    "jsx": "react",
     "sourceMap": true /* Generates corresponding '.map' file. */,
     "removeComments": true /* Do not emit comments to output. */,
-    "noEmit": true /* Do not emit outputs. */,
     "strict": true /* Enable all strict type-checking options. */,
-    "noImplicitAny": false /* Raise error on expressions and declarations with an implied 'any' type. */,
     "moduleResolution": "node" /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */,
-    "baseUrl": "./" /* Base directory to resolve non-absolute module names. */,
     "allowSyntheticDefaultImports": true /* Allow default imports from modules with no default export. This does not affect code emit, just typechecking. */,
     "forceConsistentCasingInFileNames": true /* Disallow inconsistently-cased references to the same file. */,
-    "types": [] /* Don't try to include @types packages right now. Seems to break `tsc` call */
+    "types": [] /* Don't try to include @types packages right now. Seems to break `tsc` call */,
+    "declaration": true,
+    "esModuleInterop": true,
+    "noEmit": true,
+    "noImplicitReturns": true,
+    "noImplicitThis": true,
+    "noImplicitAny": false,
+    "skipLibCheck": true,
+    "strictNullChecks": true,
+    "suppressImplicitAnyIndexErrors": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true
   },
-  "include": ["src/**/*"],
-  "exclude": ["node_modules"]
+  "include": ["src"],
+  "exclude": ["node_modules", "dist", "example", "node_modules/@types"]
```

Comment:
isn't specifying `"types": []` enough? or do we still need to exclude `node_modules/@types`?

Reply:
The `"types": []` was to fix an issue with a particular version of node and TS having issues